presto-main/src/main/java/com/facebook/presto/dispatcher/DispatchManager.java

```diff
@@ -187,11 +188,11 @@ private <C> void createQueryInternal(QueryId queryId, String slug, int retryCoun
                 throw new PrestoException(QUERY_TEXT_TOO_LARGE, format("Query text length (%s) exceeds the maximum length (%s)", queryLength, maxQueryLength));
             }
 
-            // check permissions
-            checkPermissions(queryId, sessionContext);
+            // get authorized identity if possible
+            Optional<AuthorizedIdentity> authorizedIdentity = getAuthorizedIdentity(queryId, sessionContext);
 
             // decode session
-            session = sessionSupplier.createSession(queryId, sessionContext, warningCollectorFactory);
+            session = sessionSupplier.createSession(queryId, sessionContext, warningCollectorFactory, authorizedIdentity);
```

Comment:
Do you see any issues with leveraging sessionContext to store authorizedIdentity?

Reply:
Is SessionContext logged or persisted every where. What are the components, that has access to the session context, can they take the certificate and leak it ?

Reply:
> Is SessionContext logged or persisted every where. What are the components, that has access to the session context, can they take the certificate and leak it ?

Good question! AFAIK, sessionContext is not used after the dispatch query has been created. Therefore, it should be safe to keep certificates in the context object. I just realized that it is not necessary to keep certificates in Session at all. I will remove it there. I will add another commit into this PR to reflect that.

Reply:
Done.